docs/README.md

```diff
@@ -0,0 +1,110 @@
+
+#### Limitations
+- It only supports AWS as a cloud provider.
+- It only supports Clusters using KopsControlPlane as a control plane provider.
+- A Cluster can only belong to one mesh at a time.
+
+### Security Group Controller
+#### Description
+This is the controller responsible for managing AWS Security Groups. It watches `securitygroups.ec2.aws.wildlife.io` which references Kops Operator custom resources and creates `securitygroups.ec2.aws.crossplane.io` CRs with that information. It also is responsible to attach the Security Groups to the referenced resources. 
+
+#### How to use
+Create a `securitygroups.ec2.aws.wildlife.io` CR with the desired configuration. Here's an example of a CR that creates a Security Group with two ingress rules and attaches it to two KopsMachinePools:
+```yaml
+apiVersion: ec2.aws.wildlife.io/v1alpha2
+kind: SecurityGroup
+metadata:
+  name: test-security-group
+spec:
+  infrastructureRef:
+  - apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+    kind: KopsMachinePool
+    name: nodes-a
+    namespace: k8s-test-cluster-us-east-1-general-testing
+  - apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
+    kind: KopsMachinePool
+    name: nodes-b
+    namespace: k8s-test-cluster-us-east-1-general-testing
+  ingressRules:
+  - allowedCIDRs:
+    - 0.0.0.0/0
+    fromPort: 1000
+    ipProtocol: udp
+    toPort: 1000
+```
+
+#### Diagram
+![Diagram](./security-group/sg-controller.png)
+
+#### Deep Dive
+The controller watches for changes in the Security Group CRs and creates the Crossplane CR that reflects it, after the Crossplane CR is reporting in its status that it's ready, the controller attaches them to the resources referenced in the CR. The reconciliation starts by retrieving the required information from the referenced resources they are the providerConfigName that it's going to be used as credentials in the Crossplane resource, the region and the VPC ID. The next step is to create the crossplane resource with the information retrieved and the desired configuration. After that the controller waits for the Crossplane resource to be ready by checking its status and enqueing the SecurityGroup CR again if it's not ready. When the Crossplane resource is ready the controller attaches it to the resources referenced in the SecurityGroup CR, the attachment is made in the AWS API using its SDK, both in the configuration resource like ASGs and LTs, but also in the EC2 instances to avoid drifts.
```

Comment:
`The reconciliation starts by retrieving the required information from the referenced resources, they are the providerConfigName that is going to be used as credentials in the Crossplane resource, the region, and the VPC ID.`  Do you think this is better? `The reconciliation starts by retrieving the required information from the referenced resources, they are the providerConfigName that is going to be used as credentials in the Crossplane resource, the region, and the VPC ID.
`